cms/djangoapps/contentstore/tests/test_contentstore.py

```diff
@@ -114,6 +115,96 @@ def test_xlint_fails(self):
         err_cnt = perform_xlint(TEST_DATA_DIR, ['toy'])
         self.assertGreater(err_cnt, 0)
 
+    def test_invalid_asset_name_overwrite(self):
+        """
+        Tests that an asset with invalid displayname can be overwritten.
+        During import, '/' in displayname of an asset are replaced with underscores '_'. If the converted display
+        matches with any other asset, then it will be replaced.
+        """
+        content_store = contentstore()
+        expected_displayname = '_invalid_displayname_subs-esLhHcdKGWvKs.srt'
+
+        import_course_from_xml(
+            self.store,
+            self.user.id,
+            TEST_DATA_DIR,
+            ['course_with_invalid_displayname'],
+            static_content_store=content_store,
+            verbose=True,
+            create_if_not_present=True
+        )
+
+        # Verify the course has imported successfully
+        course = self.store.get_course(self.store.make_course_key('Edx', 'EdxTest101', '2015_summer'))
+        self.assertIsNotNone(course)
+
+        # Add a new asset in the course
+        asset_key = course.id.make_asset_key('asset', 'sample_asset.srt')
+        content = StaticContent(
+            asset_key, expected_displayname, 'application/text', 'test',
+        )
+        content_store.save(content)
+
+        # Get & verify all assets of the course
+        assets, count = content_store.get_all_content_for_course(course.id)
+        self.assertEqual(count, 2)
+
+        # Verify both assets have similar 'displayname' after saving.
+        for asset in assets:
+            self.assertEquals(asset['displayname'], expected_displayname)
+
+        # Test course export is does not fail
+        root_dir = path(mkdtemp_clean())
+        print 'Exporting to tempdir = {0}'.format(root_dir)
+        export_course_to_xml(self.store, content_store, course.id, root_dir, 'test_export')
+
+        # Verify that asset have been overwritten during export.
+        filesystem = OSFS(root_dir / 'test_export/static')
+        exported_static_files = filesystem.listdir()
+        self.assertTrue(filesystem.exists(expected_displayname))
+        self.assertEqual(len(exported_static_files), 1)
+
+        # Remove exported course
+        shutil.rmtree(root_dir)
+
+    def test_asset_invalid_displayname(self):
```

Comment:
I think we are already testing this scenario in above test `test_invalid_asset_name_overwrite `  so we can remove this test. Correct me if i'm wrong.